lib/broadway_cloud_pub_sub/google_api_client.ex

```diff
@@ -55,6 +53,7 @@ defmodule BroadwayCloudPubSub.GoogleApiClient do
     %{client | adapter: adapter, pre: client.pre ++ pre}
   end
 
+  defp should_retry?({:ok, %{status: code}}), do: code in @retry_codes
   defp should_retry?({:error, %{status: code}}), do: code in @retry_codes
```

Comment:
> `defp should_retry?({:error, %{status: code}}), do: code in @retry_codes`

when is this clause called? I'd expect Tesla to return

`{:ok, %Tesla.Env{}} | {:error, reason}`

not

`{:ok, %Tesla.Env{}} | {:error, %Tesla.Env{}}`

Reply:
That makes sense. I just realized the `handle_response/2` function expects exactly that. 
I just pushed the changes.